src/main/java/ch/uzh/ifi/hase/soprafs22/entity/GameTurn.java

```diff
@@ -1,37 +1,271 @@
+            for (int possibleMove : possibleMoves) {
+                if (position + possibleMove <= 67) {
+                    return true;
+                }
+            }
+        }
+        else if (color.equals(Color.BLUE)) {
+            for (int possibleMove : possibleMoves) {
+                if (position + 28 + possibleMove <= 79) {
+                    return true;
+                }
+            }
+        }
+        else if (color.equals(Color.RED)) {
+            for (int possibleMove : possibleMoves) {
+                if (position + 52 + possibleMove <= 71) {
+                    return true;
+                }
+            }
+        }
+        else if (color.equals(Color.YELLOW)) {
+            for (int possibleMove : possibleMoves) {
+                if (position + 40 + possibleMove <= 75) {
+                    return true;
+                }
+            }
+        }
+        return false;
     }
 
-    public void makeMove(Rank cardRank) {
-        // TODO: move with a given card
+    // FOR SINGLE CARD
+    public void calculateMove(Card card) {
+
+        Rank cardRank = card.getRank();
+        if (cardRank != Rank.JOKER) {
+            updatePossibleMoves(cardRank);
+        }
+        else {
+            Rank jokerRank = chooseJokerRank();
+            updatePossibleMoves(jokerRank);
+        }
+
+        //TODO: WEBSOCKET CALL TO HIGHLIGHT POSSIBLE MOVES WITH CHOSEN CARD
+    }
+
+    public void updatePossibleDestinations () {
+        List<Integer> possibleDestinations = new ArrayList<Integer>();
+
+        // TODO: PLAYERSTATE COULD MAYBE HAVE IT'S BALLS
+//        for (Ball b: balls) {
+//            if (b.getColor().equals(activePlayer.getColor())) {
+//                for (int possibleMove : possibleMoves) {
+//                    // modulo div as board's last pos is 63
+//                    possibleDestinations.add((b.getPosition() + possibleMove) % 63);
+//                }
+//            }
+//        }
+
+        Set<Integer> set = new LinkedHashSet<>();
+        set.addAll(possibleMoves);
+        possibleMoves.clear();
+        possibleMoves.addAll(set);
+
+        setPossibleMoves(possibleMoves);
+    }
+
+    public void updatePossibleMoves(Rank cardRank) {
+        List<Integer> possibleMoves = new ArrayList<Integer>();
+
+        if (normalCards.get(cardRank) != null) {
+            // TODO: Maybe implement possibleDestinations
+            //int moveLen = normalCards.get(cardRank);
+            possibleMoves.add(normalCards.get(cardRank));
+            //possibleDestinations.add();
+        }
+        else if (cardRank.equals(Rank.FOUR)) {
+            possibleMoves.add(4);
+            possibleMoves.add(-4);
+        }
+        // TODO: implement cooperation with team member
+        // and killing on the way
+        // probably just a special number for the move
+        // which then will be processed as for loop of 1 distance moves
+        else if (cardRank.equals(Rank.SEVEN)) {
```

Comment:
make sure to consider the fact that a player can use his teamates marbles to move when playing a 7.